IFTPipeline.jl/src/feature-extraction.jl

```diff
@@ -90,3 +93,93 @@ function extractfeatures(;
     serialize(joinpath(output, "floe_props.jls"), props)
     return nothing
 end
+
+function extractfeatures_single(;
+    input::String, output::String, minarea::Int64, maxarea::Int64, features::Array{String}
+)
+    @info "Loading segmented floes from $input"
+    labeled_floes = Int.(load_labeled_img(input))
+
+    @info "Extracting features from each floe: $features"
+    props = IFTPipeline.extractfeatures(labeled_floes; minarea=minarea, maxarea=maxarea, features=features)
+
+    @info "Extracted properties:"
+    @info props
+
+    @info "Writing to $output"
+    FileIO.save(output, props)
+    return nothing
+end
+
+"""
+    load_labeled_img(path)
+
+Load an unsigned integer image from a file.
+
+See also: save_labeled_img
+"""
+function load_labeled_img(path::AbstractString)
+    image = FileIO.load(path)
+    image_reinterpreted = convert_uint_from_gray(image)
+    return image_reinterpreted
+end
+
+"""
+    save_labeled_img(image, path)
+
+Save an unsigned integer image to an image file.
+
+See also: load_labeled_img
+"""
+function save_labeled_img(image::AbstractArray{<:Integer}, path::AbstractString)
+    image_reinterpreted = convert_gray_from_uint(image)
+    FileIO.save(path, image_reinterpreted)
+    return path
+end
+
+int_to_fixedpoint_map = Dict(
+    UInt8 => N0f8,
+    Int8 => Q0f7,
+    UInt16 => N0f16,
+    Int16 => Q0f15,
+    UInt32 => N0f32,
+    Int32 => Q0f31,
+    UInt64 => N0f64,
+    Int64 => Q0f63
+)
+
+"""
+    convert_gray_from_uint(image)
+
+Convert an image from an unsigned integer format into a fixed-point Gray format.
+
+See also: convert_uint_from_gray
+"""
+
+function convert_gray_from_uint(image::AbstractArray{<:Integer})
+    img_type = eltype(image)
+    target_type = get(int_to_fixedpoint_map, img_type) do
+        error("Missing mapping for $img_type in convert_gray_from_uint")
+    end
+    image_reinterpreted = Gray.(reinterpret.(target_type, image))
+    return image_reinterpreted
+end
+
+fixedpoint_to_int_map = Dict(Gray{value} => key for (key, value) in int_to_fixedpoint_map)
```

Comment:
Why not do this within `convert_uint_from_gray` and avoid adding an extra object to the session?

Reply:
It's at the same level of abstraction as `INT_TO_FIXEDPOINT_MAP`, so I want it to be in the same level in the code.